### PR TITLE
update PoCL to cl_khr_command_buffer 0.9.4

### PR DESCRIPTION
**Pull request by franz**
implements the new SVM API calls for command buffers

### CODE REVIEW

File: lib/CL/clCreateCommandBufferKHR.c [hunk 1]
```diff
@@ -114,12 +114,15 @@ POname (clCreateCommandBufferKHR) (
...
-    memcpy (cmdbuf->properties, properties,
-            sizeof (cl_command_buffer_properties_khr) * 2 * num_properties);
+    {
+      cmdbuf->properties = (cl_command_buffer_properties_khr *)malloc (
+          (num_properties * 2 + 1)
+          * sizeof (cl_command_buffer_properties_khr));
+      memcpy (cmdbuf->properties, properties,
+              sizeof (cl_command_buffer_properties_khr)
+                  * (num_properties * 2 + 1));
+    }
```
Comment (jansol) [inline lib/CL/clCreateCommandBufferKHR.c:125]: what's the extra block for?

File: lib/CL/clEnqueueCommandBufferKHR.c [hunk 2]
```diff
@@ -208,37 +208,7 @@ POname (clEnqueueCommandBufferKHR) (cl_uint num_queues,
...
-              }
-            memcpy (node->command.memfill.pattern,
-                    cmd->command.memfill.pattern,
-                    cmd->command.memfill.pattern_size);
-            break;
-
-          default:
-            break;
-          }
+        errcode = pocl_copy_event_node (node, cmd);
```
Comment (jansol) [inline lib/CL/clEnqueueCommandBufferKHR.c:211]: This should probably also take care of the above memcpy.

**Review by jansol — APPROVED** [review]
A couple of comments, looks fine otherwise.

**Review by pjaaskel — APPROVED** [review]

**Comment by pjaaskel** [comment]
```
The following tests FAILED:
	602 - conformance_spirv_basic_math (Failed)
Errors while running CTest
```
In the LLVM 15 CI.

```
The following tests FAILED:
	366 - conformance_main_basic_math (Failed)
Error: Process completed with exit code 8.
```
LLVM 15+16 tier1.

I wonder how these are related to the PR.

File: lib/CL/clEnqueueCommandBufferKHR.c (hunk 2 above)
Comment (franz) [reply lib/CL/clEnqueueCommandBufferKHR.c:210]: fixed

File: lib/CL/clCreateCommandBufferKHR.c (hunk 1 above)
Comment (franz) [reply lib/CL/clCreateCommandBufferKHR.c:125]: to avoid having a non-NULL but unused allocation

Comment (jansol) [reply lib/CL/clCreateCommandBufferKHR.c:125]: OH, right, you did leave the `if (...)`  line in there. I must've misread the diff at first.

**Comment by franz** [comment]
@pjaaskel  that's because of commit 0c4b21e, that commit changes the test names to match ones used by rebased OpenCL-CTS (which is not yet pushed to https://github.com/pocl/OpenCL-CTS).

**Comment by franz** [comment]
reverted the OpenCL-CTS commit, can be pushed separately later.

**Comment by pjaaskel** [comment]
Still some fails.

**Pull request merged**